Apply style guide to itself (#11059)
mendix/docs: content/en/docs/community-tools/contribute-to-mendix-docs/style-guide/grammar-formatting.md

Use an alert to emphasize or call attention to information that doesn't fit into the flow of text.

In general, keep alerts short (1-2 paragraphs). If you need to write something long or complex in an alert, consider whether it can go into its own subsection with a header instead. Avoid using code blocks, bulleted lists, or numbered lists in an alert unless they are absolutely necessary.

## Ampersands (&)

Use ampersands only when describing their use in UI elements, HTML, or programming languages. Otherwise, use "and" instead of ampersands in titles, link titles, headings, and text.

For more information, see [ampersand (&)](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/a/ampersand) in the *Microsoft Style Guide*.

> Click **Save & Close**.
>
> Extract and edit the sub-microflow.

## Bold

Use bold for UI text (for example, button names and column names). Do not put UI text in quotation marks.

> > If you can't find the toolbox, you can reopen it from the **View** menu.

Reviewer: ```suggestion
> If you cannot find the toolbox, you can reopen it from the **View** menu.
```

## Contractions

Do not use contractions. Write "it is" and not "it's."

Do not use a contraction formed from a proper noun and verb (for example, "Mendix'll…").

### Cross-Reference Text

Write the full name of the cross-referenced document instead of generic text like "look at this how-to".

When cross-referencing a how-to from outside a how-to section, prefix the link with "How to" to add context.

> For more information, see *How to Create a Basic Data Layer*.

Use these phrases consistently when cross-referencing:

* "For more information, …" **+** "…see {doc}."
* "For details on …," **+** "…see {doc}."

> For details on microflow expressions, see *Triggering Logic Using Microflows*.

> For more information, see the [Troubleshooting](/refguide/install/#troubleshooting) section of *Installing Mendix Studio Pro*.
>
> For more information, see the [Installing Studio Pro Offline](/refguide/install/#offline) section below.

Do not bold or italicize text in cross-reference hyperlinks. Hyperlink formatting eliminates the need for additional emphasis.

> Only link to URLs that use HTTPS. If you're working with a URL that uses HTTP, check if there's an HTTPS version available instead. If not, consider linking to a different site.

Reviewer: ```suggestion
Only link to URLs that use HTTPS. If you are working with a URL that uses HTTP, check if there is an HTTPS version available instead. If not, consider linking to a different site.
```

> You can download it for free at [Tortoise SVN](http://tortoisesvn.tigris.org/) (choose version 1.6).

## Dashes

### En Dash {#en-dash}

Use an en dash (`–`) in definitions and [number ranges](https://learn.microsoft.com/en-us/style-guide/punctuation/dashes-hyphens). Do not capitalize the word after an en dash in a definition.

To insert an en dash, use the keyboard shortcut <kbd>Alt</kbd> + <kbd>0150</kbd> (or <kbd>Option</kbd> + <kbd>Minus sign</kbd> on a Mac).

> Reliable – this handles the sending and retrieving of changes in Subversion

> 2015–2017

Use an en dash with spaces around it to set off introductory text in list items. If you use introductory text in a list item, make sure all other items in that list have introductory text too. Bold the introductory text if it appears in the UI. For more guidance about list formatting, see [Lists](#lists).

> * **Decline** – Click this button to reject the request. You can also add a reason. After you decline the request, the submitter will receive a notification.
> * **Download** – Click this button to download the MPK file of the component.

### Em Dash

> Use an em dash (`—`) to set off a parenthetical phrase with more emphasis than parentheses provide. Don't add spaces around an em dash.

Reviewer: ```suggestion
Use an em dash (`—`) to set off a parenthetical phrase with more emphasis than parentheses provide. Do not add spaces around an em dash.
```

To insert an em dash, use the keyboard shortcut <kbd>Alt</kbd> + <kbd>0151</kbd> (or <kbd>Option</kbd> + <kbd>Shift</kbd> + <kbd>Minus sign</kbd> on a Mac).

> The information in your page—numbers, formulas, and text—is stored in the database.
>
> If you are not sure about the details, look at the illustrations in the wizard—they can help you figure out what type of connection you are using.

## Dates {#dates}

> Write dates in the format **month day, year** where:

Reviewer: ```suggestion
Write dates in the format **month day, year**, where:
```

> The reference guide topics…

## Emphasis {#emphasis}

Do not use bold for emphasis in regular documentation. An exception is that bold can be used to emphasize product names and technical terms in release notes.

> Italics is reserved for user input and strings (which should always be in English). Do not use italics in other contexts, such as emphasis; because machine translation tools may not translate italicized words.

Reviewer: ```suggestion
Italics are reserved for user input and strings (which should always be in English). Do not use italics in other contexts, such as emphasis, because machine translation tools may not translate italicized words.
```

## Literal Text

To display Markdown symbols for bold or italics without applying the formatting, wrap the text in a code snippet (``).

> `\_propertyName\_`

Use code formatting to display URLs as plain text instead of clickable links. For more details, see [URLs](#urls).

> `www.example-address.com`

## Menu Items

Capitalize menu items and bold them.

## Numbers {#numbers}

Write out numbers 1–10 (for example, "five") unless you are providing input examples (in which case, also italicize the number).

> Five customers
>
> In the **Order** field, enter *5*.

You can use ordinal numbers for positions in ranges of numbers and other technical purposes.

Write out ordinal numbers for 1–10 (for example, "fifth").

> This is the fifth example

Do not write out ordinal numbers for 11 and above. Just use the number and the suffix.

> 101st

Do not use ordinal numbers for dates (see [Dates](#dates)).

> September 25, 2023

## Person

Address the reader of your documents using the second person instead of the first person: use "you" or "your" instead of "we", "our", or "us".

> Assume your reader is the person doing the tasks that you're documenting. Use "you" instead of "user", or use "end-user" when referring to the user of a Mendix app built on the Mendix Platform.

Reviewer: ```suggestion
Assume your reader is the person doing the tasks that you are documenting. Use "you" instead of "user", or use "end-user" when referring to the user of a Mendix app built on the Mendix Platform.
```

> This example uses DeepSeek-R1.

Do not convert descriptive example statements into imperative instructions. Do not convert instructions into passive descriptions.

## Personal or Sensitive Information

In text, screenshots, and code samples, do not show any personal or sensitive information, including real names, real email addresses, profile pictures of real users, API keys, and OpenIDs. Remove or blur out this information, or replace it with fake information.

You can replace an OpenID with a random UUID that you generate using a [UUID generator](https://www.uuidtools.com/v4).

## Placeholders {#placeholders}

### Placeholders In Sample Code

> Placeholders in sample code represent values that the user must replace when they use the sample code. For inline code placeholders, use code formatting, italics, and all caps. For code blocks, use all caps and add text instructions explaining what to replace.

Reviewer: ```suggestion
Placeholders in the sample code represent values that the user must replace when they use the sample code. For inline code placeholders, use code formatting, italics, and all caps. For code blocks, use all caps and add text instructions explaining what to replace.
```

> Follow [Google's placeholder guidance](https://developers.google.com/style/placeholders): Use a short but descriptive name for the placeholder. Don't use generic text such as "input" or "placeholder". Use uppercase characters with underscore delimiters, unless that doesn't make sense in context.

Reviewer: ```suggestion
Follow [Google's placeholder guidance](https://developers.google.com/style/placeholders): Use a short but descriptive name for the placeholder. Do not use generic text such as "input" or "placeholder". Use uppercase characters with underscore delimiters, unless that does not make sense in context.
```

> When using the following command, replace `*NAMESPACE\_NAME`* with the name of your namespace.
>
> kubectl create namespace NAMESPACE\_NAME

> Don't add ornaments around sample code placeholders unless the ornaments appear in the code.

Reviewer: ```suggestion
Do not add ornaments around sample code placeholders unless the ornaments appear in the code.
```

### Placeholders in Non-Code Elements

For non-code placeholders (such as in the Studio Pro UI), add ornaments like quotes or braces if they clarify the placeholder.

> The selected page '*PAGE\_NAME*' expects an object of type '*OBJECT\_TYPE*', which is not available here.
>
> Go to the system texts editor: **App Explore** > **App '*****APP\_NAME*****'** > **System texts** > **Text**.

## Plurals in Parentheses

> Do not put optional plurals in parentheses, for example "app(s)". Instead, use either the singular form or the plural form. If you must show both forms, write the singular form, then "or" and the plural form in parentheses.

Reviewer: ```suggestion
Do not put optional plurals in parentheses, for example, "app(s)". Instead, use either the singular form or the plural form. If you must show both forms, write the singular form, then "or" and the plural form in parentheses.
```

> Register the app environment (or environments).

## Quotation Marks and Apostrophes

Use double quotation marks (") instead of single quotation marks (') when emphasizing or distinguishing terms.

Use single quotation marks (`'`) in code snippets only when they appear in the code. Do not use them to identify code; instead, use code formatting (using "`" or "```"). Use quotation marks as ornaments around placeholders only if they appear in the product UI. For more information, see [Placeholders](#placeholders).

## Serial Comma

Use the Oxford comma (also known as the serial comma).

> The app includes microflows, nanoflows, and workflows.

## Space After Period

Use a single space between a period and the first word of the next sentence.

## URLs {#urls}

Format example URLs (those not linking to Mendix or third-party sites) with Markdown code formatting. This prevents the link from appearing as a broken third-party link.

## Verb Tense

> Use present tense. The present tense is often easier to read and understand than the past or future tense. For more information, see [Verb Tense](https://docs.microsoft.com/en-us/style-guide/grammar/verbs#verb-tense) in the *Microsoft Style Guide*.

Reviewer: ```suggestion
Use the present tense. The present tense is often easier to read and understand than the past or future tense. For more information, see [Verb Tense](https://docs.microsoft.com/en-us/style-guide/grammar/verbs#verb-tense) in the *Microsoft Style Guide*.
```
